dev-requirements.txt

```diff
@@ -0,0 +1,11 @@
+# Python packages which are required/recommended to develop cocotb
+
+tox
+
+# virtualenv is used by tox; versions below 16.1.0 cause a DeprecationWarning
+# to be shown for all code which uses virtualenv, which is the case for all code
+# we run through tox. (https://github.com/pypa/virtualenv/pull/1064)
+virtualenv >= 16.1
+
+# Requirements to build the documentation
+-r documentation/requirements.txt
```

Comment:
I'm not sure about this one. I typically build the documentation with the provided Makefile, which installs the dependencies into a venv; installing them here again doesn't help.

At the same time, I know that others don't use the Makefile -- @eric-wieser @cmarqu @themperek how do you build documentation locally?

Reply:
I do use the makefile.

Reply:
This will just result in dependencies being installed twice in CI - once in the virtualenv, and once outside the virtualenv.

Reply:
Since we already use tox to manage our other dependencies, it might make sense to retire the makefile and use tox for sphinx too: https://tox.readthedocs.io/en/latest/example/documentation.html#sphinx